jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1057,7 +1057,7 @@ public void earlyEOF()
             HttpStreamOverHTTP1 stream = _stream.get();
             if (stream != null)
             {
-                BadMessageException bad = new BadMessageException("Early EOF");
+                HttpEofException bad = new HttpEofException("Early EOF");
```

Comment:
Either hardcode the `"Early EOF"` message in the exception's `getReason()` method, or return the message passed to the constructor, but let's not mix both.